### DISCUSSION TITLE
New feature - support for multiple data sources (with federation??)

### DISCUSSION THREAD

**Discussion by elsalant** [fybrik/fybrik#530] [Ideas]
Consider the case where various organizations create data.  For example, in FogProtect, various organizations will be producing videos and metadata describing the video (e.g. time of creation, URL to video, creating organization...)

Based on policy rules, different roles may be allowed access to different data repositories.  For example, there may be a rule saying Organization A can only look at data in the repository that Organization A created, but Organization B can examine all data repositories.

Ideally, M4D would create a data path for Organization A to the repository to that it created, but return to Organization B a single data path that would (somehow) federate all the disparate repositories behind it, so that Organization B can do a search on a single URI.

**Comment by simanadler** [comment]
@elsalant Interesting requirement.  @cdoron @roee88 Am I correct in understanding that this is in essence a requirement for the read module - i.e. the flight server?
